src/DynamoCore/Configuration/PathManager.cs

```diff
@@ -131,6 +131,29 @@ internal static string BuiltinPackagesDirectory
             }
         }
 
+        private static char[] invalidCharacters = null;
+        /// <summary>
+        /// List of the Windows invalid characters
+        /// </summary>
+        internal static char[] InvalidCharacters
+        {
+            get
+            {
+                if (invalidCharacters == null)
+                {
+                    invalidCharacters = PathHelper.SpecialAndInvalidCharacters();
+                }
+                return invalidCharacters;
+            }
+            set
+            {
+                if (invalidCharacters != value)
+                {
+                    invalidCharacters = value;
+                }
+            }
```

Comment:
I dont think we need the setter here. One step forward, maybe we can just remove these since we already have the static function in `PathHelper`, what do you think?